src/ray/raylet/main.cc

```diff
@@ -170,153 +170,156 @@ int main(int argc, char *argv[]) {
   RAY_CHECK_OK(gcs_client->Connect(main_service));
   std::unique_ptr<ray::raylet::Raylet> raylet;
 
-  RAY_CHECK_OK(gcs_client->Nodes().AsyncGetInternalConfig(
-      [&](::ray::Status status,
-          const boost::optional<std::string> &stored_raylet_config) {
-        RAY_CHECK_OK(status);
-        RAY_CHECK(stored_raylet_config.has_value());
-        RayConfig::instance().initialize(stored_raylet_config.get());
+  auto f = std::async(std::launch::async, [&]() {
```

Comment:
Is this necesary? can you just call `main_service` at the end?

Btw, don't bind `boost::asio::io_service::work main_work(main_service);` at the beginning.

It's very similar to [this](https://github.com/ray-project/ray/blob/ec85f874d99c2c9c44415437c91c8370e0af7c5a/src/ray/gcs/gcs_server/gcs_server.cc#L70-L82)

Reply:
`main_service` needs to be running in order to process the `GetClusterId` RPC callback. We need to have Cluster ID by this point already in order to send this RPC. 

Currently, with future:
- This will deadlock if we do not launch it in a separate thread, because it will block on the cluster ID future, which will never get started

Without future:
- This will still deadlock if we switch to a sync API because unless we start main_service, send the GetClusterID RPC, stop main_service, and continue.